src/core/server/plugins/plugins_system.ts

```diff
@@ -167,21 +169,35 @@ export class PluginsSystem {
 
   private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
     // make _cat/plugins?format=json call to the OpenSearch instance
-    const opensearchPlugins = await this.getOpenSearchPlugins(deps);
+    const opensearchInstalledPlugins = await this.getOpenSearchPlugins(deps);
     for (const pluginName of this.satupPlugins) {
       this.log.debug(`For plugin "${pluginName}"...`);
       const plugin = this.plugins.get(pluginName)!;
-      const pluginBackendDeps = new Set([...plugin.requiredOpenSearchPlugins]);
-      pluginBackendDeps.forEach((opensearchPlugin) => {
-        if (!opensearchPlugins.find((obj) => obj.component === opensearchPlugin)) {
+      const pluginOpenSearchDeps = new Set([...plugin.requiredOpenSearchPlugins]);
+      pluginOpenSearchDeps.forEach(async (depPlugin) => {
+        // add check to see if the installing Dashboards plugin version is compatible with installed OpenSearch plugin
+        if (
+          !(await this.isVersionCompatibleOSPluginInstalled(opensearchInstalledPlugins, depPlugin))
+        ) {
           this.log.warn(
-            `OpenSearch plugin "${opensearchPlugin}" is not installed on the cluster for the OpenSearch Dashboards plugin to function as expected.`
+            `OpenSearch plugin "${depPlugin.id}" is not installed on the cluster for the OpenSearch Dashboards plugin to function as expected.`
           );
         }
       });
     }
   }
 
+  private async isVersionCompatibleOSPluginInstalled(
+    opensearchInstalledPlugins: CatPluginsResponse,
+    depPlugin: CompatibleOpenSearchPluginVersions
+  ) {
+    return opensearchInstalledPlugins.find(
+      (obj) =>
+        obj.component === depPlugin.id &&
+        semver.satisfies(semver.coerce(obj.version)!.version, depPlugin.versionRange)
```

Comment:
One thing I forgot to ask this allows for the legacy `x.y.z.0` correct? It's just within the plugin manifest I still have to write it in an actual semver format?

Reply:
`semver.coerce` should take care of it.

Reply:
Yes as Miki said `semver.coerce` will take care of trimming/translating string to semver if possible.